tools/backups2datalad.py

```diff
@@ -611,79 +439,240 @@ def describe_dandiset(self, dandiset_id):
         )
         log.debug("Got bucket URL for asset")
         return urlunparse(urlbits._replace(query=f"versionId={s3meta['VersionId']}"))
 
-    @property
-    def s3client(self):
-        if self._s3client is None:
-            self._s3client = boto3.client(
-                "s3", config=Config(signature_version=UNSIGNED)
+    def tag_releases(self, dandiset: RemoteDandiset, ds: Dataset, push: bool) -> None:
+        log.info("Tagging releases for Dandiset %s", dandiset.identifier)
+        for v in dandiset.get_versions():
+            if readcmd("git", "tag", "-l", v.identifier):
+                log.debug("Version %s already tagged", v.identifier)
+            else:
+                log.info("Tagging version %s", v.identifier)
+                self.mkrelease(dandiset.for_version(v), ds, push=push)
+
+    def mkrelease(
+        self,
+        dandiset: RemoteDandiset,
+        ds: Dataset,
+        push: bool,
+        commitish: Optional[str] = None,
+    ) -> None:
+        # `dandiset` must have its version set to the published version
+        def git(*args: str, **kwargs: Any) -> None:
+            log.debug("Running: git %s", " ".join(shlex.quote(str(a)) for a in args))
+            subprocess.run(["git", *args], cwd=repo, check=True, **kwargs)
+
+        repo: Path = ds.pathobj
+        if commitish is None:
+            commitish = readcmd(
+                "git",
+                "rev-list",
+                f"--before={dandiset.version.created}",
+                "-n1",
+                "HEAD",
+                cwd=repo,
+            )
+        remote_assets = list(dandiset.get_assets())
+        repo_assets = json.loads(
+            readcmd("git", "show", f"{commitish}:.dandi/assets.json", cwd=repo)
+        )
+        if (
+            repo_assets
+            and (
+                not isinstance(repo_assets[0], dict) or "asset_id" not in repo_assets[0]
+            )
+        ) or not assets_eq(remote_assets, repo_assets):
+            log.info(
+                "Assets in commit %s do not match assets in version %s;"
+                " creating branch for version",
+                commitish,
+                dandiset.version_id,
             )
-        return self._s3client
-
-    @property
-    def gh(self):
-        if self._gh is None:
-            token = subprocess.check_output(
-                ["git", "config", "hub.oauthtoken"], universal_newlines=True
-            ).strip()
-            self._gh = Github(token)
-        return self._gh
-
-    @staticmethod
-    def mklink(src, dest):
-        log.info("cp %s -> %s", src, dest)
-        subprocess.run(
-            [
-                "cp",
-                "-L",
-                "--reflink=always",
-                "--remove-destination",
-                str(src),
-                str(dest),
-            ],
-            check=True,
+            branching = True
+            git("checkout", "-b", f"release-{dandiset.version_id}")
```

Comment:
was there some specific motivation to tag/branch for `draft`, or just an omission of an exclusion?
```
(dandisets) dandi@drogon:/mnt/backup/dandi/tmp/dandisets$ git -C 000029 tag
0.210712.1903
0.210730.1538
0.210805.2047
0.210806.1511
0.210806.2112
draft
(dandisets) dandi@drogon:/mnt/backup/dandi/tmp/dandisets$ git -C 000029 branch
  git-annex
* master
  release-0.210712.1903
  release-0.210730.1538
  release-0.210805.2047
  release-0.210806.1511
  release-draft
```
IMHO there should be no `draft` tag or branch (master is a `draft`... we could make that explicit I guess, but that would be a separate issue)

Reply:
I also wonder if it is worth keeping release branches... any benefit? as long as we have tags, those could be recreated, and they would not give user any advantage.  I would just delete them upon successfull minting of a tag

Reply:
@yarikoptic "draft" is no longer tagged.  I think having release branches makes the history a bit easier to navigate.

Reply:
> @yarikoptic "draft" is no longer tagged.
cool!

> I think having release branches makes the history a bit easier to navigate.

I wonder *how*, if they would be either on the main branch or just a single commit "off" the main branch and have a tag there?

Reply:
`release-*` branches are now deleted after creating the tag.